File: common/src/main/java/com/hartwig/actin/common/DatamodelPrinter.java
package com.hartwig.actin.common;

import com.google.common.base.Strings;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.jetbrains.annotations.NotNull;

public class DatamodelPrinter {

    private static final Logger LOGGER = LogManager.getLogger(DatamodelPrinter.class);
    private static final int DEFAULT_INDENTATION = 1;

    private final int indentation;

    @NotNull
    public static DatamodelPrinter withDefaultIndentation() {
        return new DatamodelPrinter(DEFAULT_INDENTATION);
    }

    public DatamodelPrinter(final int indentation) {
        this.indentation = indentation;
    }

    public void print(@NotNull String line) {
        LOGGER.info("{}{}", Strings.repeat(" ", indentation), line);
    }
}

File: common/src/main/kotlin/com/hartwig/actin/util/DatamodelPrinter.kt
package com.hartwig.actin.util

import io.github.oshai.kotlinlogging.KotlinLogging

class DatamodelPrinter(private val indentation: Int) {

    fun print(line: String) {
        logger.info { "${" ".repeat(indentation)}$line" }
    }

    companion object {
        private val logger = KotlinLogging.logger {}
        private const val DEFAULT_INDENTATION: Int = 1

        fun withDefaultIndentation(): DatamodelPrinter {
            return DatamodelPrinter(DEFAULT_INDENTATION)
        }
    }
}
